### PR TITLE
Feature/add reward

### PR DESCRIPTION
**Pull request by thebluber**
- Add reward resource
- Bump version to 1.5.0 (which also includes the multisite support)

### CODE REVIEW

**Review by fractaledmind — APPROVED** [review]

File: spec/cirro_io/client/reward_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                type: 'app-workers',
+                id: 1,
+              },
+            },
+          },
+          attributes: attributes,
+        },
+      }
+
+      reward = described_class.create(**attributes.merge(app_worker: app_worker))
```
Comment (softpro) [inline spec/cirro_io/client/reward_spec.rb:37]: `**attributes, app_worker: app_worker` should work, I guess 🤔

File: spec/cirro_io/client/reward_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    it 'creates a reward for an app worker' do
+      request_url = "#{test_site}/v1/rewards"
+      stub_request(:post, request_url)
+        .to_return(body: File.read('./spec/fixtures/reward.json'), headers: { 'Content-Type' => 'application/json' })
+
+      request_body = {
+        data: {
+          type: 'rewards',
```
Comment (softpro) [inline spec/cirro_io/client/reward_spec.rb:24]: `bonus-payouts`

File: lib/cirro_io/client/reward.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module CirroIO
+  module Client
+    class Reward < Base
```
Comment (softpro) [inline lib/cirro_io/client/reward.rb:3]: It looks like object type is tied to the URL here 🤔 The type is `bonus-payouts` in fact. Or we should rename it ASAP.

**Review by softpro — CHANGES_REQUESTED** [review]
The name inconsistency goes worse 😕

**Comment by thebluber** [comment]
put this on hold until we figure out the issue with `bonus-payouts` vs `rewards` on the api side.

**Comment by softpro** [comment]
Depends on [cirro#357](https://github.com/test-IO/cirro/pull/357).

File: lib/cirro_io/client/reward.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+      has_one :app_worker
```
Comment (softpro) [inline lib/cirro_io/client/reward.rb]: It `belongs_to :app_worker`, actually. Therefore, it requests `/v1/app-workers/:app_worker_id/payouts` by default. Thus, we should alter the endpoint on backend again 🤦

**Review by softpro — APPROVED** [review]

File: lib/cirro_io/client/reward.rb (hunk 4 above)
Comment (softpro) [reply lib/cirro_io/client/reward.rb]: I've tried to further modify [cirro#360](https://github.com/test-IO/cirro/pull/360) to get the desired behavior using `JSONAPI::Resources`, but haven't succeeded yet.

Comment (thebluber) [reply lib/cirro_io/client/reward.rb]: I don't understand you here. Why do you assume that it requests `/v1/app-workers/:app_worker_id/payouts` by default? I don't think so. 
Take a look at client/gig.rb it is the same there with `worker_filter` and it is working fine with `/v1/gigs`

**Review by softpro — APPROVED** [review]

**Pull request merged**